discord-impl/src/main/java/net/starype/quiz/discordimpl/core/BotMain.java

```diff
@@ -1,39 +1,26 @@
 package net.starype.quiz.discordimpl.core;
 
-import discord4j.core.DiscordClientBuilder;
-import discord4j.core.GatewayDiscordClient;
-import discord4j.core.event.EventDispatcher;
-import discord4j.core.event.domain.message.MessageCreateEvent;
-import discord4j.core.event.domain.message.ReactionAddEvent;
+import net.dv8tion.jda.api.JDA;
+import net.dv8tion.jda.api.JDABuilder;
+import net.dv8tion.jda.api.entities.Activity;
+import net.dv8tion.jda.api.utils.cache.CacheFlag;
 import net.starype.quiz.discordimpl.game.GameList;
 import net.starype.quiz.discordimpl.game.LobbyList;
 import net.starype.quiz.discordimpl.input.MessageInputListener;
 import net.starype.quiz.discordimpl.input.ReactionInputListener;
 
-import java.io.BufferedReader;
-import java.io.FileReader;
-import java.io.IOException;
-
 public class BotMain {
 
-    public static void main(String[] args) {
-        GatewayDiscordClient client = DiscordClientBuilder.create(System.getenv("BOT_TOKEN"))
-                .build()
-                .login()
-                .block();
-        ReactionInputListener reactionListener = new ReactionInputListener();
-        EventDispatcher dispatcher = client.getEventDispatcher();
+    public static void main(String[] args) throws Exception {
 
-        dispatcher
-                .on(ReactionAddEvent.class)
-                .filter(ReactionInputListener.createFilter())
-                .retry()
-                .subscribe(reactionListener);
-        dispatcher
-                .on(MessageCreateEvent.class)
-                .filter(MessageInputListener.createFilter())
-                .retry()
-                .subscribe(new MessageInputListener(new LobbyList(reactionListener), new GameList()));
-        client.onDisconnect().block();
+        JDABuilder builder = JDABuilder.createDefault(System.getenv("BOT_TOKEN"));
+        JDA jda = builder.disableCache(CacheFlag.MEMBER_OVERRIDES, CacheFlag.VOICE_STATE)
+                .setBulkDeleteSplittingEnabled(false)
+                .setActivity(Activity.playing("supervising the quiz world championship"))

```

Comment:
This name is illegal

Reply:
"supervising the quiz world championship"

Reply:
So is your nickname

Reply:
This PR could have been merged before 5 month